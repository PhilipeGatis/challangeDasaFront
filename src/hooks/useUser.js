import useAxios from 'axios-hooks';

export default (name) => {
  const [{ data, loading, error }, refetch] = useAxios(`https://api.github.com/users/${name}`);
  const hasError = error && error.response && !!error.response.status;
  return [{ data, loading, error: hasError }, refetch];
};
