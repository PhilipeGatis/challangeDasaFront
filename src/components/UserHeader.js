import React from 'react';
import { Avatar } from 'antd';
import { Typography, Space, Button, Tooltip } from 'antd';
import useUser from '../hooks/useUser';
import styled from 'styled-components';
import ArrowLeftOutlined from '@ant-design/icons/ArrowLeftOutlined';
import { useHistory, useParams } from 'react-router-dom';

const { Text } = Typography;

const StyleText = styled(Text)`
  color: #fff;
`;

const StyleButton = styled(Button)`
  color: #fff;
`;

const UserHeader = () => {
  const { user } = useParams();
  const history = useHistory();
  const [{ data, loading, error }] = useUser(user);

  if (error) {
    const errorMessage = `error=User not found: ${user}`;
    history.push({
      pathname: '/',
      search: errorMessage,
    });
    return null;
  }

  if (loading) return null;
  return (
    <Space direction="horizontal">
      <Tooltip title="Back">
        <StyleButton type="link" id="back" href="/" shape="circle" icon={<ArrowLeftOutlined />} />
      </Tooltip>
      <Avatar size={40} src={data.avatar_url} />
      <StyleText>{data.name || data.login}</StyleText>
    </Space>
  );
};

export default UserHeader;
