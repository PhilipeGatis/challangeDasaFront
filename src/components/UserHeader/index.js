import React from 'react';
import { Avatar } from 'antd';
import { Typography, Space, Button, Tooltip } from 'antd';
import useUser from '../hooks/useUser';
import styled from 'styled-components';
import ArrowLeftOutlined from '@ant-design/icons/ArrowLeftOutlined';

const { Text } = Typography;

const StyleText = styled(Text)`
  color: #fff;
`;

const StyleButton = styled(Button)`
  color: #fff;
`;

const UserHeader = ({ history, match }) => {
  const { user } = match.params;
  const [{ data, loading, error }] = useUser(user);

  if (error) {
    const errorMessage = `error=User not found: ${user}`;
    history.push({
      pathname: '/',
      search: errorMessage,
    });
    return null;
  }

  if (loading) return null;
  return (
    <Space direction="horizontal">
      <Tooltip title="Back">
        <StyleButton type="link" href="/" shape="circle" icon={<ArrowLeftOutlined />} />
      </Tooltip>
      <Avatar size={40} src={data.avatar_url} />
      <StyleText>{data.name}</StyleText>
    </Space>
  );
};

export default UserHeader;
