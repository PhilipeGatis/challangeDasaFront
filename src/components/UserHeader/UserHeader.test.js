import React from 'react';
import UserHeader from './UserHeader';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import useUser from '../../hooks/useUser';
import userJSON from '../../fixtures/user';

jest.mock('../../hooks/useUser');

describe('<UserHeader />', () => {
  const RenderWithRouter = ({ children }) => (
    <MemoryRouter initialEntries={['/PhilipeGatis']}>
      <Route path="/:user">{children}</Route>
    </MemoryRouter>
  );

  it('should render', () => {
    useUser.mockReturnValue([
      {
        data: userJSON,
        loading: false,
      },
    ]);
    const wrapper = render(
      <RenderWithRouter>
        <UserHeader />
      </RenderWithRouter>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
