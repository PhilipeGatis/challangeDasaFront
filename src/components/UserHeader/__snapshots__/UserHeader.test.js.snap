// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserHeader /> should render 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center"
>
  <div
    class="ant-space-item"
    style="margin-right:8px"
  >
    <a
      class="ant-btn sc-AxirZ eFUgaj ant-btn-link ant-btn-circle ant-btn-icon-only"
      href="/"
      id="back"
    >
      <span
        aria-label="arrow-left"
        class="anticon anticon-arrow-left"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="arrow-left"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </a>
  </div>
  <div
    class="ant-space-item"
    style="margin-right:8px"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image"
      style="width:40px;height:40px;line-height:40px;font-size:18px"
    >
      <img
        src="/avatar.jpg"
      />
    </span>
  </div>
  <div
    class="ant-space-item"
  >
    <span
      class="ant-typography sc-AxjAm iyJsvQ"
    >
      Philipe Gatis
    </span>
  </div>
</div>
`;
